File: java-backport/net/jakedot/jourwigo/openwig/EventTable.java
package net.jakedot.jourwigo.openwig;

import java.util.HashMap;
import java.util.Map;

public class EventTable {
    public Map<String, Object> table = new HashMap<>();
    public Object position = null;

    public void setPosition(Object pos) {
        this.position = pos;
    }

    public boolean hasEvent(String eventName) {
        // In a real Kahlua environment, this checks if the table contains a LuaClosure
        return table.containsKey(eventName) && table.get(eventName) != null;
    }

    public Object callEvent(String eventName, Object[] args) {
        if (hasEvent(eventName)) {
            // In Kahlua, this would invoke the Lua closure.
            // This is a stub for the backport.
            System.out.println("Calling event: " + eventName);
        }
        return null;
    }

    public void setItem(String key, Object value) {
        table.put(key, value);
    }

    public Object getItem(String key) {
        return table.get(key);
    }
}

File: src/lib/openwig/EventTable.ts
export class EventTable {
  public table: Record<string, any> = {};
  public position: any = null;

  public setPosition(pos: any) {
    this.position = pos;
  }

  public hasEvent(eventName: string): boolean {
    return typeof this.table[eventName] === 'function';
  }

  public callEvent(eventName: string, args: any[] = []): any {
    if (this.hasEvent(eventName)) {
      return this.table[eventName](this.table, ...args);
    }
    return null;
  }

  public setItem(key: string, value: any) {
    this.table[key] = value;
  }

  public getItem(key: string): any {
    return this.table[key];
  }
}
